htlcswitch/link.go

```diff
@@ -528,6 +534,28 @@ func (l *channelLink) Stop() {
 	}
 }
 
+func (l *channelLink) Flush(onFlushed func()) error {
+	if !l.flushCont.TryPut(onFlushed) {
+		return errors.New(
+			"can't flush because flush already in progress",
+		)
+	}
+
+	return nil
+}
+
+func (l *channelLink) CancelFlush() error {
+	if l.flushCont.TryTake().IsNone() {
```

Comment:
Does this later on add the mechanics to _resume_ normal channel execution? If not, then what's gained by allowing a flush to. be cancelled?

Reply:
There may be a discussion to have here. In this case since all channel link impairment is implicit on the state of this MVar the cancel will implicitly resume normal channel function. The only "irrevocable" action is the shutdown of the htlc manager which happens in the continuation inserted here. So it can be cancelled until the flushCont begins executing. We can remove this API though, as it was only added for completeness. I can't think of a reason en-protocol that we want this.